packages/auto-approve/src/check-config.ts

```diff
@@ -0,0 +1,151 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This file manages the logic to check whether a given config file is valid
+
+// eslint-disable-next-line node/no-extraneous-import
+import {ProbotOctokit} from 'probot';
+import {join} from 'path';
+import yaml from 'js-yaml';
+import Ajv from 'ajv';
+const ajv = new Ajv();
+
+const CONFIGURATION_FILE_PATH = 'auto-approve.yml';
+
+interface File {
+  content: string | undefined;
+}
+
+export interface ErrorMessage {
+  wrongProperty: Record<string, string>;
+  message: string | undefined;
+}
+const schema = require(join(
+  __dirname,
+  '../',
+  '../',
+  'src',
+  'valid-pr-schema.json'
+));
+
+function isFile(file: File | unknown): file is File {
+  return (file as File).content !== undefined;
+}
+
+/**
+ * Takes in the auto-approve.yml file and checks to see that it is formatted correctly
+ *
+ * @param configYaml the string of auto-approve.yml
+ * @returns true if it is a valid YAML object, an error message if it is not valid
+ */
+export function validateYaml(configYaml: string): Boolean | string {
+  try {
+    const isYaml = yaml.load(configYaml);
+    if (typeof isYaml === 'object') {
+      return true;
+    } else {
+      return 'File is not a YAML object';
+    }
+  } catch (err) {
+    return 'File is not properly configured YAML';
+  }
+}
+
+/**
+ * Takes in the auto-approve.yml file and checks to see that the schema matches the valid-pr-schema.json rules
+ *
+ * @param configYaml the string of auto-approve.yml
+ * @returns true if it matches valid-pr-schema.json, the error messages if it doesn't match
+ */
+export async function validateSchema(
```

Comment:
I agree 👍  I'd be tempted to throw an Error instead, which could have the additional contextual information (which you can catch at the calling function).

Then I'd just make this resolve `Boolean` 😄

Reply:
Sounds good! Thanks for the thoughts :)

Reply:
So, I considered returning an error but that seemed to be a lot of lines of code... so instead I'm returning a standardized validation object, that looks like this: 

```
export interface validationResult {
  checkType: 'Yaml' | 'Schema' | 'Codeowners';
  isValid: boolean;
  errorMessages?: ErrorMessage[];
  message?: string | undefined;
}
```
Feel free to advocate for an error though, if I'm missing something!

Reply:
`validationResult` still makes it complicated to call this function.  To understand the the result, I have to find the definition of this interface, and then write code that inspects the 4 members of the result.

I propose the following function interface:

```js
/**
 * Takes in the auto-approve.yml file and checks to see that it is formatted correctly
 *
 * @param configYaml the string of auto-approve.yml
 * @returns empty if the yaml is valid, otherwise an error message.
 */
export function validateYaml(configYaml: string): string;
```

With this declaration, it's very easy to understand the return value and hard to get the logic wrong.

Reply:
Awesome, thanks for the comments! I added this in.

Reply:
I still see validateYaml returning `string | undefined` on line 53.

Reply:
I just made the changes we talked about last week :)

Reply:
I see.